Traceback (most recent call last):
  File "/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 166, in wrapped
    return loop.run_until_complete(inner)
  File "/usr/lib/python3.10/asyncio/base_events.py", line 646, in run_until_complete
    return future.result()
  File "/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
from mp_api.client import MPRester
import os
import pymatgen as pmg

from ase.io import write

MPID = 'mp-20674' # Materials Project ID number

"""
  The 'with ...' statement defines an MPRester code block. Subsequent
  indented statements belong to the code block, and the object mpr
  may be used within the code block.
"""
with MPRester(MP_API_KEY) as mpr:
    # Get only the structure for YBa2Cu3O7
    structure = mpr.get_structure_by_material_id(MPID)
------------------

----- stderr -----
/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/mp_api/client/mprester.py:182: UserWarning: mpcontribs-client not installed. Install the package to query MPContribs data, or construct pourbaix diagrams: 'pip install mpcontribs-client'
  warnings.warn(
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mgaierror[0m                                  Traceback (most recent call last)
File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/urllib3/connection.py:200[0m, in [0;36mHTTPConnection._new_conn[0;34m(self)[0m
[1;32m    199[0m [38;5;28;01mtry[39;00m:
[0;32m--> 200[0m     sock [38;5;241m=[39m [43mconnection[49m[38;5;241;43m.[39;49m[43mcreate_connection[49m[43m([49m
[1;32m    201[0m [43m        [49m[43m([49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_dns_host[49m[43m,[49m[43m [49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mport[49m[43m)[49m[43m,[49m
[1;32m    202[0m [43m        [49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mtimeout[49m[43m,[49m
[1;32m    203[0m [43m        [49m[43msource_address[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43msource_address[49m[43m,[49m
[1;32m    204[0m [43m        [49m[43msocket_options[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43msocket_options[49m[43m,[49m
[1;32m    205[0m [43m    [49m[43m)[49m
[1;32m    206[0m [38;5;28;01mexcept[39;00m socket[38;5;241m.[39mgaierror [38;5;28;01mas[39;00m e:

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/urllib3/util/connection.py:60[0m, in [0;36mcreate_connection[0;34m(address, timeout, source_address, socket_options)[0m
[1;32m     58[0m     [38;5;28;01mraise[39;00m LocationParseError([38;5;124mf[39m[38;5;124m"[39m[38;5;124m'[39m[38;5;132;01m{[39;00mhost[38;5;132;01m}[39;00m[38;5;124m'[39m[38;5;124m, label empty or too long[39m[38;5;124m"[39m) [38;5;28;01mfrom[39;00m [38;5;28;01mNone[39;00m
[0;32m---> 60[0m [38;5;28;01mfor[39;00m res [38;5;129;01min[39;00m [43msocket[49m[38;5;241;43m.[39;49m[43mgetaddrinfo[49m[43m([49m[43mhost[49m[43m,[49m[43m [49m[43mport[49m[43m,[49m[43m [49m[43mfamily[49m[43m,[49m[43m [49m[43msocket[49m[38;5;241;43m.[39;49m[43mSOCK_STREAM[49m[43m)[49m:
[1;32m     61[0m     af, socktype, proto, canonname, sa [38;5;241m=[39m res

File [0;32m/usr/lib/python3.10/socket.py:955[0m, in [0;36mgetaddrinfo[0;34m(host, port, family, type, proto, flags)[0m
[1;32m    954[0m addrlist [38;5;241m=[39m []
[0;32m--> 955[0m [38;5;28;01mfor[39;00m res [38;5;129;01min[39;00m [43m_socket[49m[38;5;241;43m.[39;49m[43mgetaddrinfo[49m[43m([49m[43mhost[49m[43m,[49m[43m [49m[43mport[49m[43m,[49m[43m [49m[43mfamily[49m[43m,[49m[43m [49m[38;5;28;43mtype[39;49m[43m,[49m[43m [49m[43mproto[49m[43m,[49m[43m [49m[43mflags[49m[43m)[49m:
[1;32m    956[0m     af, socktype, proto, canonname, sa [38;5;241m=[39m res

[0;31mgaierror[0m: [Errno -3] Temporary failure in name resolution

The above exception was the direct cause of the following exception:

[0;31mNameResolutionError[0m                       Traceback (most recent call last)
File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/urllib3/connectionpool.py:790[0m, in [0;36mHTTPConnectionPool.urlopen[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)[0m
[1;32m    789[0m [38;5;66;03m# Make the request on the HTTPConnection object[39;00m
[0;32m--> 790[0m response [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_make_request[49m[43m([49m
[1;32m    791[0m [43m    [49m[43mconn[49m[43m,[49m
[1;32m    792[0m [43m    [49m[43mmethod[49m[43m,[49m
[1;32m    793[0m [43m    [49m[43murl[49m[43m,[49m
[1;32m    794[0m [43m    [49m[43mtimeout[49m[38;5;241;43m=[39;49m[43mtimeout_obj[49m[43m,[49m
[1;32m    795[0m [43m    [49m[43mbody[49m[38;5;241;43m=[39;49m[43mbody[49m[43m,[49m
[1;32m    796[0m [43m    [49m[43mheaders[49m[38;5;241;43m=[39;49m[43mheaders[49m[43m,[49m
[1;32m    797[0m [43m    [49m[43mchunked[49m[38;5;241;43m=[39;49m[43mchunked[49m[43m,[49m
[1;32m    798[0m [43m    [49m[43mretries[49m[38;5;241;43m=[39;49m[43mretries[49m[43m,[49m
[1;32m    799[0m [43m    [49m[43mresponse_conn[49m[38;5;241;43m=[39;49m[43mresponse_conn[49m[43m,[49m
[1;32m    800[0m [43m    [49m[43mpreload_content[49m[38;5;241;43m=[39;49m[43mpreload_content[49m[43m,[49m
[1;32m    801[0m [43m    [49m[43mdecode_content[49m[38;5;241;43m=[39;49m[43mdecode_content[49m[43m,[49m
[1;32m    802[0m [43m    [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mresponse_kw[49m[43m,[49m
[1;32m    803[0m [43m[49m[43m)[49m
[1;32m    805[0m [38;5;66;03m# Everything went great![39;00m

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/urllib3/connectionpool.py:491[0m, in [0;36mHTTPConnectionPool._make_request[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)[0m
[1;32m    490[0m         new_e [38;5;241m=[39m _wrap_proxy_error(new_e, conn[38;5;241m.[39mproxy[38;5;241m.[39mscheme)
[0;32m--> 491[0m     [38;5;28;01mraise[39;00m new_e
[1;32m    493[0m [38;5;66;03m# conn.request() calls http.client.*.request, not the method in[39;00m
[1;32m    494[0m [38;5;66;03m# urllib3.request. It also calls makefile (recv) on the socket.[39;00m

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/urllib3/connectionpool.py:467[0m, in [0;36mHTTPConnectionPool._make_request[0;34m(self, conn, method, url, body, headers, retries, timeout, chunked, response_conn, preload_content, decode_content, enforce_content_length)[0m
[1;32m    466[0m [38;5;28;01mtry[39;00m:
[0;32m--> 467[0m     [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_validate_conn[49m[43m([49m[43mconn[49m[43m)[49m
[1;32m    468[0m [38;5;28;01mexcept[39;00m (SocketTimeout, BaseSSLError) [38;5;28;01mas[39;00m e:

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/urllib3/connectionpool.py:1092[0m, in [0;36mHTTPSConnectionPool._validate_conn[0;34m(self, conn)[0m
[1;32m   1091[0m [38;5;28;01mif[39;00m conn[38;5;241m.[39mis_closed:
[0;32m-> 1092[0m     [43mconn[49m[38;5;241;43m.[39;49m[43mconnect[49m[43m([49m[43m)[49m
[1;32m   1094[0m [38;5;28;01mif[39;00m [38;5;129;01mnot[39;00m conn[38;5;241m.[39mis_verified:

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/urllib3/connection.py:604[0m, in [0;36mHTTPSConnection.connect[0;34m(self)[0m
[1;32m    603[0m sock: socket[38;5;241m.[39msocket [38;5;241m|[39m ssl[38;5;241m.[39mSSLSocket
[0;32m--> 604[0m [38;5;28mself[39m[38;5;241m.[39msock [38;5;241m=[39m sock [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43m_new_conn[49m[43m([49m[43m)[49m
[1;32m    605[0m server_hostname: [38;5;28mstr[39m [38;5;241m=[39m [38;5;28mself[39m[38;5;241m.[39mhost

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/urllib3/connection.py:207[0m, in [0;36mHTTPConnection._new_conn[0;34m(self)[0m
[1;32m    206[0m [38;5;28;01mexcept[39;00m socket[38;5;241m.[39mgaierror [38;5;28;01mas[39;00m e:
[0;32m--> 207[0m     [38;5;28;01mraise[39;00m NameResolutionError([38;5;28mself[39m[38;5;241m.[39mhost, [38;5;28mself[39m, e) [38;5;28;01mfrom[39;00m [38;5;21;01me[39;00m
[1;32m    208[0m [38;5;28;01mexcept[39;00m SocketTimeout [38;5;28;01mas[39;00m e:

[0;31mNameResolutionError[0m: <urllib3.connection.HTTPSConnection object at 0x7f4f320218a0>: Failed to resolve 'api.materialsproject.org' ([Errno -3] Temporary failure in name resolution)

The above exception was the direct cause of the following exception:

[0;31mMaxRetryError[0m                             Traceback (most recent call last)
File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/requests/adapters.py:486[0m, in [0;36mHTTPAdapter.send[0;34m(self, request, stream, timeout, verify, cert, proxies)[0m
[1;32m    485[0m [38;5;28;01mtry[39;00m:
[0;32m--> 486[0m     resp [38;5;241m=[39m [43mconn[49m[38;5;241;43m.[39;49m[43murlopen[49m[43m([49m
[1;32m    487[0m [43m        [49m[43mmethod[49m[38;5;241;43m=[39;49m[43mrequest[49m[38;5;241;43m.[39;49m[43mmethod[49m[43m,[49m
[1;32m    488[0m [43m        [49m[43murl[49m[38;5;241;43m=[39;49m[43murl[49m[43m,[49m
[1;32m    489[0m [43m        [49m[43mbody[49m[38;5;241;43m=[39;49m[43mrequest[49m[38;5;241;43m.[39;49m[43mbody[49m[43m,[49m
[1;32m    490[0m [43m        [49m[43mheaders[49m[38;5;241;43m=[39;49m[43mrequest[49m[38;5;241;43m.[39;49m[43mheaders[49m[43m,[49m
[1;32m    491[0m [43m        [49m[43mredirect[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m,[49m
[1;32m    492[0m [43m        [49m[43massert_same_host[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m,[49m
[1;32m    493[0m [43m        [49m[43mpreload_content[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m,[49m
[1;32m    494[0m [43m        [49m[43mdecode_content[49m[38;5;241;43m=[39;49m[38;5;28;43;01mFalse[39;49;00m[43m,[49m
[1;32m    495[0m [43m        [49m[43mretries[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mmax_retries[49m[43m,[49m
[1;32m    496[0m [43m        [49m[43mtimeout[49m[38;5;241;43m=[39;49m[43mtimeout[49m[43m,[49m
[1;32m    497[0m [43m        [49m[43mchunked[49m[38;5;241;43m=[39;49m[43mchunked[49m[43m,[49m
[1;32m    498[0m [43m    [49m[43m)[49m
[1;32m    500[0m [38;5;28;01mexcept[39;00m (ProtocolError, [38;5;167;01mOSError[39;00m) [38;5;28;01mas[39;00m err:

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/urllib3/connectionpool.py:844[0m, in [0;36mHTTPConnectionPool.urlopen[0;34m(self, method, url, body, headers, retries, redirect, assert_same_host, timeout, pool_timeout, release_conn, chunked, body_pos, preload_content, decode_content, **response_kw)[0m
[1;32m    842[0m     new_e [38;5;241m=[39m ProtocolError([38;5;124m"[39m[38;5;124mConnection aborted.[39m[38;5;124m"[39m, new_e)
[0;32m--> 844[0m retries [38;5;241m=[39m [43mretries[49m[38;5;241;43m.[39;49m[43mincrement[49m[43m([49m
[1;32m    845[0m [43m    [49m[43mmethod[49m[43m,[49m[43m [49m[43murl[49m[43m,[49m[43m [49m[43merror[49m[38;5;241;43m=[39;49m[43mnew_e[49m[43m,[49m[43m [49m[43m_pool[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[43m,[49m[43m [49m[43m_stacktrace[49m[38;5;241;43m=[39;49m[43msys[49m[38;5;241;43m.[39;49m[43mexc_info[49m[43m([49m[43m)[49m[43m[[49m[38;5;241;43m2[39;49m[43m][49m
[1;32m    846[0m [43m[49m[43m)[49m
[1;32m    847[0m retries[38;5;241m.[39msleep()

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/urllib3/util/retry.py:515[0m, in [0;36mRetry.increment[0;34m(self, method, url, response, error, _pool, _stacktrace)[0m
[1;32m    514[0m     reason [38;5;241m=[39m error [38;5;129;01mor[39;00m ResponseError(cause)
[0;32m--> 515[0m     [38;5;28;01mraise[39;00m MaxRetryError(_pool, url, reason) [38;5;28;01mfrom[39;00m [38;5;21;01mreason[39;00m  [38;5;66;03m# type: ignore[arg-type][39;00m
[1;32m    517[0m log[38;5;241m.[39mdebug([38;5;124m"[39m[38;5;124mIncremented Retry for (url=[39m[38;5;124m'[39m[38;5;132;01m%s[39;00m[38;5;124m'[39m[38;5;124m): [39m[38;5;132;01m%r[39;00m[38;5;124m"[39m, url, new_retry)

[0;31mMaxRetryError[0m: HTTPSConnectionPool(host='api.materialsproject.org', port=443): Max retries exceeded with url: /heartbeat (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x7f4f320218a0>: Failed to resolve 'api.materialsproject.org' ([Errno -3] Temporary failure in name resolution)"))

During handling of the above exception, another exception occurred:

[0;31mConnectionError[0m                           Traceback (most recent call last)
Cell [0;32mIn[3], line 14[0m
[1;32m      7[0m MPID [38;5;241m=[39m [38;5;124m'[39m[38;5;124mmp-20674[39m[38;5;124m'[39m [38;5;66;03m# Materials Project ID number[39;00m
[1;32m      9[0m [38;5;124;03m"""[39;00m
[1;32m     10[0m [38;5;124;03m  The 'with ...' statement defines an MPRester code block. Subsequent[39;00m
[1;32m     11[0m [38;5;124;03m  indented statements belong to the code block, and the object mpr[39;00m
[1;32m     12[0m [38;5;124;03m  may be used within the code block.[39;00m
[1;32m     13[0m [38;5;124;03m"""[39;00m
[0;32m---> 14[0m [38;5;28;01mwith[39;00m [43mMPRester[49m[43m([49m[43mMP_API_KEY[49m[43m)[49m [38;5;28;01mas[39;00m mpr:
[1;32m     15[0m     [38;5;66;03m# Get only the structure for YBa2Cu3O7[39;00m
[1;32m     16[0m     structure [38;5;241m=[39m mpr[38;5;241m.[39mget_structure_by_material_id(MPID)

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/mp_api/client/mprester.py:192[0m, in [0;36mMPRester.__init__[0;34m(self, api_key, endpoint, notify_db_version, include_user_agent, monty_decode, use_document_model, session, headers)[0m
[1;32m    189[0m     warnings[38;5;241m.[39mwarn([38;5;124mf[39m[38;5;124m"[39m[38;5;124mProblem loading MPContribs client: [39m[38;5;132;01m{[39;00merror[38;5;132;01m}[39;00m[38;5;124m"[39m)
[1;32m    191[0m [38;5;66;03m# Check if emmet version of server os compatible[39;00m
[0;32m--> 192[0m emmet_version [38;5;241m=[39m version[38;5;241m.[39mparse([38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mget_emmet_version[49m[43m([49m[43m)[49m)
[1;32m    194[0m [38;5;28;01mif[39;00m version[38;5;241m.[39mparse(emmet_version[38;5;241m.[39mbase_version) [38;5;241m<[39m version[38;5;241m.[39mparse(
[1;32m    195[0m     _MAPI_SETTINGS[38;5;241m.[39mMIN_EMMET_VERSION
[1;32m    196[0m ):
[1;32m    197[0m     warnings[38;5;241m.[39mwarn(
[1;32m    198[0m         [38;5;124m"[39m[38;5;124mThe installed version of the mp-api client may not be compatible with the API server. [39m[38;5;124m"[39m
[1;32m    199[0m         [38;5;124m"[39m[38;5;124mPlease install a previous version if any problems occur.[39m[38;5;124m"[39m
[1;32m    200[0m     )

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/mp_api/client/mprester.py:340[0m, in [0;36mMPRester.get_emmet_version[0;34m(self)[0m
[1;32m    333[0m [38;5;28;01mdef[39;00m [38;5;21mget_emmet_version[39m([38;5;28mself[39m):
[1;32m    334[0m [38;5;250m    [39m[38;5;124;03m"""[39;00m
[1;32m    335[0m [38;5;124;03m    Get the latest version emmet-core and emmet-api used in the[39;00m
[1;32m    336[0m [38;5;124;03m    current API service.[39;00m
[1;32m    337[0m 
[1;32m    338[0m [38;5;124;03m    Returns: version as a string[39;00m
[1;32m    339[0m [38;5;124;03m    """[39;00m
[0;32m--> 340[0m     [38;5;28;01mreturn[39;00m [43mget[49m[43m([49m[43murl[49m[38;5;241;43m=[39;49m[38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43mendpoint[49m[43m [49m[38;5;241;43m+[39;49m[43m [49m[38;5;124;43m"[39;49m[38;5;124;43mheartbeat[39;49m[38;5;124;43m"[39;49m[43m)[49m[38;5;241m.[39mjson()[[38;5;124m"[39m[38;5;124mversion[39m[38;5;124m"[39m]

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/requests/api.py:73[0m, in [0;36mget[0;34m(url, params, **kwargs)[0m
[1;32m     62[0m [38;5;28;01mdef[39;00m [38;5;21mget[39m(url, params[38;5;241m=[39m[38;5;28;01mNone[39;00m, [38;5;241m*[39m[38;5;241m*[39mkwargs):
[1;32m     63[0m [38;5;250m    [39m[38;5;124mr[39m[38;5;124;03m"""Sends a GET request.[39;00m
[1;32m     64[0m 
[1;32m     65[0m [38;5;124;03m    :param url: URL for the new :class:`Request` object.[39;00m
[0;32m   (...)[0m
[1;32m     70[0m [38;5;124;03m    :rtype: requests.Response[39;00m
[1;32m     71[0m [38;5;124;03m    """[39;00m
[0;32m---> 73[0m     [38;5;28;01mreturn[39;00m [43mrequest[49m[43m([49m[38;5;124;43m"[39;49m[38;5;124;43mget[39;49m[38;5;124;43m"[39;49m[43m,[49m[43m [49m[43murl[49m[43m,[49m[43m [49m[43mparams[49m[38;5;241;43m=[39;49m[43mparams[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/requests/api.py:59[0m, in [0;36mrequest[0;34m(method, url, **kwargs)[0m
[1;32m     55[0m [38;5;66;03m# By using the 'with' statement we are sure the session is closed, thus we[39;00m
[1;32m     56[0m [38;5;66;03m# avoid leaving sockets open which can trigger a ResourceWarning in some[39;00m
[1;32m     57[0m [38;5;66;03m# cases, and look like a memory leak in others.[39;00m
[1;32m     58[0m [38;5;28;01mwith[39;00m sessions[38;5;241m.[39mSession() [38;5;28;01mas[39;00m session:
[0;32m---> 59[0m     [38;5;28;01mreturn[39;00m [43msession[49m[38;5;241;43m.[39;49m[43mrequest[49m[43m([49m[43mmethod[49m[38;5;241;43m=[39;49m[43mmethod[49m[43m,[49m[43m [49m[43murl[49m[38;5;241;43m=[39;49m[43murl[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/requests/sessions.py:587[0m, in [0;36mSession.request[0;34m(self, method, url, params, data, headers, cookies, files, auth, timeout, allow_redirects, proxies, hooks, stream, verify, cert, json)[0m
[1;32m    582[0m send_kwargs [38;5;241m=[39m {
[1;32m    583[0m     [38;5;124m"[39m[38;5;124mtimeout[39m[38;5;124m"[39m: timeout,
[1;32m    584[0m     [38;5;124m"[39m[38;5;124mallow_redirects[39m[38;5;124m"[39m: allow_redirects,
[1;32m    585[0m }
[1;32m    586[0m send_kwargs[38;5;241m.[39mupdate(settings)
[0;32m--> 587[0m resp [38;5;241m=[39m [38;5;28;43mself[39;49m[38;5;241;43m.[39;49m[43msend[49m[43m([49m[43mprep[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43msend_kwargs[49m[43m)[49m
[1;32m    589[0m [38;5;28;01mreturn[39;00m resp

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/requests/sessions.py:701[0m, in [0;36mSession.send[0;34m(self, request, **kwargs)[0m
[1;32m    698[0m start [38;5;241m=[39m preferred_clock()
[1;32m    700[0m [38;5;66;03m# Send the request[39;00m
[0;32m--> 701[0m r [38;5;241m=[39m [43madapter[49m[38;5;241;43m.[39;49m[43msend[49m[43m([49m[43mrequest[49m[43m,[49m[43m [49m[38;5;241;43m*[39;49m[38;5;241;43m*[39;49m[43mkwargs[49m[43m)[49m
[1;32m    703[0m [38;5;66;03m# Total elapsed time of the request (approximately)[39;00m
[1;32m    704[0m elapsed [38;5;241m=[39m preferred_clock() [38;5;241m-[39m start

File [0;32m/media/colin/Shared/colin/git/materials-ml-workshop/env/lib/python3.10/site-packages/requests/adapters.py:519[0m, in [0;36mHTTPAdapter.send[0;34m(self, request, stream, timeout, verify, cert, proxies)[0m
[1;32m    515[0m     [38;5;28;01mif[39;00m [38;5;28misinstance[39m(e[38;5;241m.[39mreason, _SSLError):
[1;32m    516[0m         [38;5;66;03m# This branch is for urllib3 v1.22 and later.[39;00m
[1;32m    517[0m         [38;5;28;01mraise[39;00m SSLError(e, request[38;5;241m=[39mrequest)
[0;32m--> 519[0m     [38;5;28;01mraise[39;00m [38;5;167;01mConnectionError[39;00m(e, request[38;5;241m=[39mrequest)
[1;32m    521[0m [38;5;28;01mexcept[39;00m ClosedPoolError [38;5;28;01mas[39;00m e:
[1;32m    522[0m     [38;5;28;01mraise[39;00m [38;5;167;01mConnectionError[39;00m(e, request[38;5;241m=[39mrequest)

[0;31mConnectionError[0m: HTTPSConnectionPool(host='api.materialsproject.org', port=443): Max retries exceeded with url: /heartbeat (Caused by NameResolutionError("<urllib3.connection.HTTPSConnection object at 0x7f4f320218a0>: Failed to resolve 'api.materialsproject.org' ([Errno -3] Temporary failure in name resolution)"))

